### DISCUSSION TITLE
How to control the fan and temperature?

### DISCUSSION THREAD

**Discussion by hobus** [DomiStyle/esphome-panasonic-ac#129] [General]
I have been using this program with my Panasonic HZ25ZKE Air-conditioner through Home assistant and directly through ESPhome and it works great.
But I have yet to find out how I can change temperature (selected heating) and fan speed either with a Home Assistant automation or via a http call. For example I can turn on/off Nanoex with this call: http://192.168.0.85/switch/panasonic_ac_nanoex_switch/turn_on
And I can change temp/fan speed with the HA UI, but can not find how to call these settings via yaml or a http call.
Can anyone please hint me how to?

**Comment by mathiasschenker** [comment]
I would also like to be able to set the target temperature via a script in home assistant. As of now the target temperature can only be controlled with the dial, but the value is not an entity. If I understand correctly, that would be required for this purpose.

**Reply by frassefraze** [reply]
Have a look at the Climate integration in the home assistant docs, in the temperature case climate.set_temperature
 https://www.home-assistant.io/integrations/climate/

**Reply by mathiasschenker** [reply]
thanks, that was exactly what I was looking for!
